### PR TITLE
Setup dialyzer

### PR DESCRIPTION
**Pull request by jdenen**
This PR runs dialyzer in Travis, but it only verifies that we adhere to contracts/behaviors. It ignores errors related to failing function calls, failing lambda calls, and functions with no local returns.

I generally like dialyzer. @bbalser generally doesn't. We felt that this was the best compromise to bring it into the project.

Travis is set up to cache the master branch build's `.plt` directory, so it's going to take awhile to generate the PLT for this PR and its merge to master.

### CODE REVIEW

File: lib/elsa/topic.ex [hunk 1]
```diff
@@ -71,6 +71,7 @@ defmodule Elsa.Topic do
     end)
   end
 
+  @spec send_request(pid(), term(), integer()) :: :ok | {:ok, term()} | {:error, term()}
```
Comment (jeffgrunewald) [inline lib/elsa/topic.ex]: do you want to add a spec to this private function?

**Comment by jeffgrunewald** [comment]
Initial build with dialyzer seems to have timed out; assuming we're sure about including dialyzer are we sure we want it to be part of the Travis build?

**Comment by jdenen** [comment]
Travis build shouldn't be an issue once it caches the PLT. Getting to that point appears to be a PITA, though.

File: lib/elsa/topic.ex (hunk 1 above)
Comment (jdenen) [reply lib/elsa/topic.ex]: Nice catch. I think this was me debugging issues and was mistakenly left behind.

**Comment by jdenen** [comment]
Also, if we don't include it in CI, then there's no point in having it. No one will run it.

File: lib/elsa/util.ex [hunk 2]
```diff
@@ -102,6 +102,7 @@ defmodule Elsa.Util do
     |> hd()
   end
 
+  @spec connect(list(), atom()) :: {:ok, pid()} | {:error, term()}
```
Comment (jeffgrunewald) [inline lib/elsa/util.ex]: another private function

**Pull request merged**